commands/intent.go

```diff
@@ -0,0 +1,29 @@
+package commands
+
+const (
+	// DefaultIntent Default Intent Type
+	DefaultIntent = iota
+	// VersionIntent Intent to print version
+	VersionIntent = iota
+)
+
+// Intent Used for deriving user intent based on predefined business rules
+type Intent struct {
+	PrintVersion bool
+}
+
+// NewIntent Intent Constructor
+func NewIntent(printVersion bool) Intent {
+	return Intent{
+		PrintVersion: printVersion,
+	}
+}
+
+// Is Get derived intent type
+func (i *Intent) Is() int {
```

Comment:
Is `int` the correct return type here? Can we be more specific?

Reply:
Yah, [you can use a typealias](https://github.com/daltonclaybrook/dalton-website/blob/75ae1b889e6d9d8809cc86a5143c24a203e8a770/lambda/workouts/workouts.go#L233-L242). It's not actually any more _type safe_ than what you have, though.

Reply:
Does that show up in tooling at least? That at least seems more descriptive.

Reply:
Yes